src/main/scala/io/iohk/ethereum/ledger/Ledger.scala

```diff
@@ -16,7 +17,7 @@ object Ledger extends Logger {
     storages: BlockchainStorages,
     stateStorage: NodeStorage): Unit = {
 
-    val blockError = validateBlockBeforeExecution(block)
+    val blockError = validateBlockBeforeExecution(block, BlockchainImpl(storages))
```

Comment:
Blockchain is also being created in `io.iohk.ethereum.ledger.Ledger#executeBlockTransactions` maybe it's a good idea to use this one as parameter on that method too